auth_pysaml_website/controllers/main.py

```diff
@@ -0,0 +1,18 @@
+# Copyright (C) 2020 GlodoUK <https://www.glodo.uk/>
+
+from odoo.http import request
+
+from odoo.addons.web.controllers.main import Home
+
+
+class SAMLLogin(Home):
+    def _list_saml_providers_domain(self):
+        website_id = getattr(request, "website", None)
+        if website_id:
+            return [
+                "|",
+                ("website_ids.id", "=", website_id.id),
+                ("website_ids", "=", False),
+            ]
```

Comment:
You should also validate that a key is set, otherwise if the attachment is removed or the key is cleared, the provider will be listed but an error will occurr when clicking it:

```suggestion
                            ("sp_pem", "!=", False
                        ]
```